### ISSUE TITLE
HTTP ERROR 503 Service Unavailable

### ISSUE THREAD

**Issue by mfide** [scm-manager/scm-manager#2214]
Hello,

I use scm about a year using docker image `scmmanager/scm-manager:latest`. (Host system is Ubuntu 22.04 and CPU: AMD Phenom II X4 960T Processor)

Yesterday watchtower (auto docker image update tool) detected there was a new image and updated the docker container automatically to the latest one.

This is an update log from watchtower:
```
time="2024-05-30T04:00:52Z" level=info msg="Found new scmmanager/scm-manager:latest image (a7f85c2f4de4)"
time="2024-05-30T04:01:47Z" level=info msg="Stopping /svn-server (774d0d5ae585) with SIGTERM"
time="2024-05-30T04:01:49Z" level=info msg="Creating /svn-server"
```

After this update, I cannot open the scm web page anymore, it says:

```
HTTP ERROR 503 Service Unavailable
URI:	/scm/
STATUS:	503
MESSAGE:	Service Unavailable
SERVLET:	-
```

Do you have any idea why it could be? Or Where do you suggest me to check?

Here is my docker-compose.yml:
```
services:
  scm:
    image: scmmanager/scm-manager:latest
    container_name: svn-server
    volumes:
      - ./scmhome:/var/lib/scm
    restart: unless-stopped
    environment:
      - SCM_FORWARD_HEADERS_ENABLED=true
```

And also here is the content of container log:
```
2024-05-30 12:59:19.181:INFO::main: Logging initialized @257ms to org.eclipse.jetty.util.log.StdErrLog
2024-05-30 12:59:19.502:INFO:oejs.Server:main: jetty-9.4.51.v20230217; built: 2023-02-17T08:19:37.309Z; git: b45c405e4544384de066f814ed42ae3dceacdd49; jvm 17.0.11+9-alpine-r0
2024-05-30 12:59:21.469:INFO:oejw.StandardDescriptorProcessor:main: NO JSP Support for /scm, did not find org.eclipse.jetty.jsp.JettyJspServlet
2024-05-30 12:59:21.498:INFO:oejs.session:main: DefaultSessionIdManager workerName=node0
2024-05-30 12:59:21.498:INFO:oejs.session:main: No SessionScavenger set, using defaults
2024-05-30 12:59:21.502:INFO:oejs.session:main: node0 Scavenging every 600000ms
2024-05-30 12:59:21.821 [main] [          ] INFO  sonia.scm.lifecycle.classloading.ClassLoaderLifeCycle - create new simple ClassLoaderLifeCycle
2024-05-30 12:59:21.826 [main] [          ] INFO  sonia.scm.lifecycle.BootstrapContextFilter - register for restart events
2024-05-30 12:59:21.833 [main] [          ] INFO  sonia.scm.event.LegmanScmEventBus - create new event bus ScmEventBus-1
2024-05-30 12:59:21.954 [main] [          ] WARN  io.micrometer.core.instrument.binder.jvm.ExecutorServiceMetrics - Failed to bind as org.apache.shiro.concurrent.SubjectAwareExecutorService is unsupported.
2024-05-30 12:59:21.956 [main] [          ] INFO  sonia.scm.event.ScmEventBus - use sonia.scm.event.LegmanScmEventBus as event bus implementation
2024-05-30 12:59:22.025 [main] [          ] INFO  sonia.scm.lifecycle.BootstrapContextListener - start scm-manager initialization
2024-05-30 12:59:23.149 [main] [          ] INFO  sonia.scm.plugin.PluginProcessor - collect plugins
2024-05-30 12:59:23.342 [main] [          ] INFO  sonia.scm.plugin.PluginProcessor - install plugin tree:
+- scm-repository-avatar-plugin@1.2.1
+- scm-binary-search-plugin@1.0.1
+- scm-trace-monitor-plugin@1.1.0
+- scm-activity-plugin@2.0.0
+- scm-audit-log-plugin@1.0.0
+- scm-git-plugin@2.46.3
+- scm-legacy-plugin@2.46.3
+- scm-commit-search-plugin@1.0.0
+- scm-hg-plugin@2.46.3
+- scm-content-search-plugin@1.2.1
   +- scm-binary-search-plugin@1.0.1
+- scm-readme-plugin@2.1.0
+- scm-code-editor-plugin@1.1.0
   +- scm-editor-plugin@2.12.1
+- scm-editor-plugin@2.12.1
+- scm-landingpage-plugin@1.14.0
   +- scm-trace-monitor-plugin@1.1.0
+- scm-repository-size-plugin@1.0.0
+- scm-mail-plugin@2.10.1
   +- scm-notify-plugin@2.0.4
+- scm-gravatar-plugin@2.0.0
+- scm-svn-plugin@2.46.3
+- scm-el-plugin@1.1.0
+- scm-statistic-plugin@2.2.2
+- scm-notify-plugin@2.0.4

2024-05-30 12:59:23.461:WARN:oejw.WebAppContext:main: Failed startup of context o.e.j.w.WebAppContext@392a04e7{SCM-Manager ${project.version},/scm,file:///var/cache/scm/work/webapp/webapp/,UNAVAILABLE}{./var/webapp/scm-webapp.war}
java.lang.NoClassDefFoundError: jakarta/servlet/ServletContextListener
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
        at java.base/java.security.SecureClassLoader.defineClass(Unknown Source)
        at java.base/java.net.URLClassLoader.defineClass(Unknown Source)
        at java.base/java.net.URLClassLoader$1.run(Unknown Source)
        at java.base/java.net.URLClassLoader$1.run(Unknown Source)
        at java.base/java.security.AccessController.doPrivileged(Unknown Source)
        at java.base/java.net.URLClassLoader.findClass(Unknown Source)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        at com.sun.xml.bind.v2.model.impl.RuntimeBuiltinLeafInfoImpl$9.parse(RuntimeBuiltinLeafInfoImpl.java:342)
        at com.sun.xml.bind.v2.model.impl.RuntimeBuiltinLeafInfoImpl$9.parse(RuntimeBuiltinLeafInfoImpl.java:332)
        at com.sun.xml.bind.v2.runtime.unmarshaller.TextLoader.text(TextLoader.java:39)
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.text(UnmarshallingContext.java:558)
        at com.sun.xml.bind.v2.runtime.unmarshaller.SAXConnector.processText(SAXConnector.java:182)
        at com.sun.xml.bind.v2.runtime.unmarshaller.SAXConnector.endElement(SAXConnector.java:145)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.endElement(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanEndElement(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:228)
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:199)
        at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:140)
        at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:179)
        at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:164)
        at sonia.scm.plugin.PluginProcessor.createDescriptor(PluginProcessor.java:458)
        at sonia.scm.plugin.PluginProcessor.createPlugin(PluginProcessor.java:474)
        at sonia.scm.plugin.PluginProcessor.appendPluginWrapper(PluginProcessor.java:287)
        at sonia.scm.plugin.PluginProcessor.createPluginWrappers(PluginProcessor.java:506)
        at sonia.scm.plugin.PluginProcessor.collectPlugins(PluginProcessor.java:187)
        at sonia.scm.plugin.PluginsInternal.collectPlugins(PluginsInternal.java:62)
        at sonia.scm.lifecycle.PluginBootstrap.collectPlugins(PluginBootstrap.java:108)
        at sonia.scm.lifecycle.PluginBootstrap.<init>(PluginBootstrap.java:79)
        at sonia.scm.lifecycle.BootstrapContextListener.createStageTwoInjector(BootstrapContextListener.java:115)
        at sonia.scm.lifecycle.BootstrapContextListener.getInjector(BootstrapContextListener.java:93)
        at com.google.inject.servlet.GuiceServletContextListener.contextInitialized(GuiceServletContextListener.java:45)
        at sonia.scm.lifecycle.BootstrapContextListener.contextInitialized(BootstrapContextListener.java:75)
        at sonia.scm.lifecycle.BootstrapContextFilter.initializeContext(BootstrapContextFilter.java:74)
        at sonia.scm.lifecycle.BootstrapContextFilter.init(BootstrapContextFilter.java:65)
        at org.eclipse.jetty.servlet.FilterHolder.initialize(FilterHolder.java:140)
        at org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0(ServletHandler.java:750)
        at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Unknown Source)
        at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(Unknown Source)
        at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:774)
        at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:379)
        at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1449)
        at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1414)
        at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:916)
        at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:288)
        at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:524)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
        at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
        at org.eclipse.jetty.server.Server.start(Server.java:423)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:110)
        at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
        at org.eclipse.jetty.server.Server.doStart(Server.java:387)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
        at sonia.scm.server.ScmServer.init(ScmServer.java:105)
        at sonia.scm.server.ScmServer.run(ScmServer.java:66)
        at sonia.scm.server.ScmServerDaemon.main(ScmServerDaemon.java:53)
Caused by: 
java.lang.ClassNotFoundException: jakarta.servlet.ServletContextListener
        at java.base/java.net.URLClassLoader.findClass(Unknown Source)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(Unknown Source)
        at java.base/java.security.SecureClassLoader.defineClass(Unknown Source)
        at java.base/java.net.URLClassLoader.defineClass(Unknown Source)
        at java.base/java.net.URLClassLoader$1.run(Unknown Source)
        at java.base/java.net.URLClassLoader$1.run(Unknown Source)
        at java.base/java.security.AccessController.doPrivileged(Unknown Source)
        at java.base/java.net.URLClassLoader.findClass(Unknown Source)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        at java.base/java.lang.ClassLoader.loadClass(Unknown Source)
        at com.sun.xml.bind.v2.model.impl.RuntimeBuiltinLeafInfoImpl$9.parse(RuntimeBuiltinLeafInfoImpl.java:342)
        at com.sun.xml.bind.v2.model.impl.RuntimeBuiltinLeafInfoImpl$9.parse(RuntimeBuiltinLeafInfoImpl.java:332)
        at com.sun.xml.bind.v2.runtime.unmarshaller.TextLoader.text(TextLoader.java:39)
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.text(UnmarshallingContext.java:558)
        at com.sun.xml.bind.v2.runtime.unmarshaller.SAXConnector.processText(SAXConnector.java:182)
        at com.sun.xml.bind.v2.runtime.unmarshaller.SAXConnector.endElement(SAXConnector.java:145)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.endElement(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanEndElement(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl$FragmentContentDriver.next(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentScannerImpl.next(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLNSDocumentScannerImpl.next(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XML11Configuration.parse(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.XMLParser.parse(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.parsers.AbstractSAXParser.parse(Unknown Source)
        at java.xml/com.sun.org.apache.xerces.internal.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source)
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal0(UnmarshallerImpl.java:228)
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallerImpl.unmarshal(UnmarshallerImpl.java:199)
        at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:140)
        at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:179)
        at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:164)
        at sonia.scm.plugin.PluginProcessor.createDescriptor(PluginProcessor.java:458)
        at sonia.scm.plugin.PluginProcessor.createPlugin(PluginProcessor.java:474)
        at sonia.scm.plugin.PluginProcessor.appendPluginWrapper(PluginProcessor.java:287)
        at sonia.scm.plugin.PluginProcessor.createPluginWrappers(PluginProcessor.java:506)
        at sonia.scm.plugin.PluginProcessor.collectPlugins(PluginProcessor.java:187)
        at sonia.scm.plugin.PluginsInternal.collectPlugins(PluginsInternal.java:62)
        at sonia.scm.lifecycle.PluginBootstrap.collectPlugins(PluginBootstrap.java:108)
        at sonia.scm.lifecycle.PluginBootstrap.<init>(PluginBootstrap.java:79)
        at sonia.scm.lifecycle.BootstrapContextListener.createStageTwoInjector(BootstrapContextListener.java:115)
        at sonia.scm.lifecycle.BootstrapContextListener.getInjector(BootstrapContextListener.java:93)
        at com.google.inject.servlet.GuiceServletContextListener.contextInitialized(GuiceServletContextListener.java:45)
        at sonia.scm.lifecycle.BootstrapContextListener.contextInitialized(BootstrapContextListener.java:75)
        at sonia.scm.lifecycle.BootstrapContextFilter.initializeContext(BootstrapContextFilter.java:74)
        at sonia.scm.lifecycle.BootstrapContextFilter.init(BootstrapContextFilter.java:65)
        at org.eclipse.jetty.servlet.FilterHolder.initialize(FilterHolder.java:140)
        at org.eclipse.jetty.servlet.ServletHandler.lambda$initialize$0(ServletHandler.java:750)
        at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Unknown Source)
        at java.base/java.util.stream.Streams$ConcatSpliterator.forEachRemaining(Unknown Source)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(Unknown Source)
        at org.eclipse.jetty.servlet.ServletHandler.initialize(ServletHandler.java:774)
        at org.eclipse.jetty.servlet.ServletContextHandler.startContext(ServletContextHandler.java:379)
        at org.eclipse.jetty.webapp.WebAppContext.startWebapp(WebAppContext.java:1449)
        at org.eclipse.jetty.webapp.WebAppContext.startContext(WebAppContext.java:1414)
        at org.eclipse.jetty.server.handler.ContextHandler.doStart(ContextHandler.java:916)
        at org.eclipse.jetty.servlet.ServletContextHandler.doStart(ServletContextHandler.java:288)
        at org.eclipse.jetty.webapp.WebAppContext.doStart(WebAppContext.java:524)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:117)
        at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:169)
        at org.eclipse.jetty.server.Server.start(Server.java:423)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:110)
        at org.eclipse.jetty.server.handler.AbstractHandler.doStart(AbstractHandler.java:97)
        at org.eclipse.jetty.server.Server.doStart(Server.java:387)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
        at sonia.scm.server.ScmServer.init(ScmServer.java:105)
        at sonia.scm.server.ScmServer.run(ScmServer.java:66)
        at sonia.scm.server.ScmServerDaemon.main(ScmServerDaemon.java:53)
2024-05-30 12:59:23.477:INFO:oejw.StandardDescriptorProcessor:main: NO JSP Support for /, did not find org.eclipse.jetty.jsp.JettyJspServlet
2024-05-30 12:59:23.479:INFO:oejsh.ContextHandler:main: Started o.e.j.w.WebAppContext@6b587673{/,[file:///opt/scm-server/var/webapp/docroot/],AVAILABLE}
2024-05-30 12:59:23.517:INFO:oejs.AbstractConnector:main: Started ServerConnector@436a4e4b{HTTP/1.1, (http/1.1)}{0.0.0.0:8080}
2024-05-30 12:59:23.517:INFO:oejs.Server:main: Started @4600ms
```

Your helps are highly appreciated.
Thank you.

**Comment by pfeuffer** [comment]
Hi @mfide , we had a misconfiguration in our deployment and an older version became `latest`. We will release the correct version asap. Sorry for the inconvenience!

**Comment by pfeuffer** [comment]
Hi @mfide again, the `latest` tag is fixed again and not links to version 3.2.1.

A big sorry again for the issue and thank you for the hint!

**Comment by mfide** [comment]
Thank you, I updated the docker image again using following commands and all good now: `docker compose pull && docker compose up -d`
